p2p/src/connection/webrtc/outgoing/p2p_connection_webrtc_outgoing_state.rs

```diff
@@ -5,70 +5,64 @@ use serde::{Deserialize, Serialize};
 
 use openmina_core::requests::RpcId;
 
-use crate::{connection::RejectionReason, webrtc};
+use crate::{
+    connection::{webrtc::RejectionReason, ConnectionState},
+    webrtc,
+};
 
-use super::P2pConnectionOutgoingInitOpts;
-
-#[derive(Serialize, Deserialize, Debug, Clone)]
-pub enum P2pConnectionOutgoingState {
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub enum P2pConnectionWebRTCOutgoingState {
+    #[default]
+    Default,
     Init {
         time: Timestamp,
-        opts: P2pConnectionOutgoingInitOpts,
```

Comment:
Because we already have address in the state. Why duplicate it?
```rust
pub struct P2pWebRTCPeerState {
    pub dial_opts: Option<SignalingMethod>,
    pub status: P2pWebRTCPeerStatus,
}
```

Reply:
Because it's a matter of when we set it. We set `dial_addr` on a successful connection, or if it's empty, if I'm not mistaken. Maybe that's no longer the case.

My knowledge of p2p is probably fairly outdated now, so if the above won't pose any problems, then it's fine I guess.